core/util/http.py

```diff
@@ -191,6 +191,9 @@ def request_with_timeout(
             url, sessions.Session.request, http_method, *args, **kwargs
         )
 
+    # The set of status codes on which a retry will be attempted (if the number of retries requested is non-zero).
+    RETRY_STATUS_CODES = list(range(500, 600))
```

Comment:
Minor: This is fine, but we can probably be a bit more precise with it. 

I'd suggest: 
```python
RETRY_STATUS_CODES = [429, 500, 502, 503, 504]
```